### PR TITLE
Add a controller address validation

### PR DESCRIPTION
**Pull request by honza**
Validate the controller address in the create a new baremetal host dialog

### CODE REVIEW

File: src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js [hunk 1]
```diff
@@ -17,6 +18,7 @@ const formFields = {
     id: 'controller',
     title: 'Management Controller Address',
     required: true,
+    validate: form => validateBmcURL(form.controller.value),
```
Comment (rawagner) [inline src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js]: you can use `settingsValue(form, 'controller')`

Comment (honza) [reply src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js]: That seems like an awfully specific (i.e. not general) function.

File: src/utils/validations.js [hunk 2]
```diff
@@ -110,3 +112,38 @@ export const validateVmwareURL = value => {
...
+export const validateBmcURL = value => {
+  if (!value) {
+    return getValidationObject(EMPTY_ERROR);
+  }
+
+  const hasProtocol = value.includes('://');
+  const hasPort = value.match(':(?!/)');
+
+  if (hasProtocol) {
+    const startsWithAllowedProtocol = startsWith(value, 'ipmi') || startsWith(value, 'idrac');
```
Comment (suomiy) [inline src/utils/validations.js]: nitpick: could be also done this way
 
```js
if(hasProtocol && !['impi', 'idrac'].find(allowedProtocol => startsWith(value, allowedProtocol)){
  return getValidationObject(BMC_PROTOCOL_ERROR);
}
```

and `['impi', 'idrac']` could be  extracted to a constant.

File: src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js (hunk 1 above)
Comment (rawagner) [reply src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js]: it is actually meant to be used with `FormFactory` so it applies here too. Problem is that we need to change function name and name of the parameters and move it to better place to make it seem more generic :) I will refactor it.

File: src/utils/validations.js [hunk 3]
```diff
@@ -110,3 +112,38 @@ export const validateVmwareURL = value => {
...
+
+    if (!startsWithAllowedProtocol) {
+      return getValidationObject(BMC_PROTOCOL_ERROR);
+    }
+  }
+
+  if (hasPort) {
```
[the remark this answers is not in the document]
Comment (honza) [reply src/utils/validations.js]: Done

File: src/utils/validations.js (hunk 2 above)
Comment (honza) [reply src/utils/validations.js]: Done

File: src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js (hunk 1 above)
Comment (honza) [reply src/components/CreateBaremetalHostDialog/CreateBaremetalHostDialog.js]: Done, I'm happy to update it once it's refactored.

**Pull request merged**